src/Simulation/Simulators.Tests/Circuits/UserDefinedTypes.qs

```diff
@@ -7,10 +7,11 @@ namespace Microsoft.Quantum.Simulation.Simulators.Tests.Circuits
 
     newtype P1 = (Int, Int);
     newtype P2 = ((Int, Int), Int);
+    newtype NamedTuple = (FirstItem: (Int, Double), SecondItem: Int);
```

Comment:
Looks good. If you could please add one more test for an inner named tuple item (it's all working fine, but never hurts to cover it with a test): 
```suggestion
    newtype NamedTuple = (FirstItem: (Int, Double), SecondItem: Int);
    newtype P3 = ((Fst : Result, Snd : (Int, Int)), Trd : String);

    function UdtInnerNamedTupleFieldTest () : Unit {
        let t = P3((Zero, (0,1)), "");
        AssertEqual(Zero, t::Fst);
        let snd = t::Snd;
        let (s1, s2) = snd;
        AssertEqual(0, s1);
        AssertEqual(1, s2);
        AssertEqual("", t::Trd);
    }
```

Reply:
Added. Thanks!